iceoryx_dust/include/iceoryx_dust/internal/cxx/std_string_support.inl

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_STD_STRING_SUPPORT_INL
+#define IOX_DUST_STD_STRING_SUPPORT_INL
+
+#include "iceoryx_dust/cxx/std_string_support.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+template <uint64_t N>
+inline std::string FromImpl<string<N>, std::string>::fromImpl(const string<N>& value)
+{
+    return std::string(value.c_str(), value.size());
+}
+
+template <uint64_t N>
+inline string<N> FromImpl<std::string, string<N>>::fromImpl(const std::string& value) noexcept
+{
+    return string<N>(TruncateToCapacity, value.c_str(), value.size());
+}
```

Comment:
What do you think about this in addition to the implementation which truncates
```cpp
template <uint64_t N>
inline optional<string<N>> FromImpl<std::string, optional<string<N>>>::fromImpl(const std::string& value) noexcept
{
    auto stringSize = value.size();
    if (stringSize > N) {
        return nullopt;
    }
    return string<N>(TruncateToCapacity, value.c_str(), stringSize);
}
```

cc @elfenpiff @MatthiasKillat

Reply:
I like it!

Reply:
@elBoberido Neat idea. I like it, too! However, as this PR already has an approval and is already quite larger I would implement this on 2nd PR.

Reply:
@mossmaurice the more I think about it, the more I come to the conclusion that the plain `string<N> FromImpl<std::string, string<N>>::fromImpl` should be removed since it silently truncates without the indication that it happens. If you agree, this could be done on a follow up PR, to move this to dust.

Reply:
@elBoberido In general, I would say it makes sense for the code parts which take user inputs. Anywhere, were fixed strings are used or the length is known in advance I think we don't need it. I'll have a look after the merge, ok?

Reply:
@mossmaurice I think I have a solution that lets me sleep peacefully

Let's assume we have this
```cpp
template <typename D, typename R>
struct Lossy {
    using Destination = D;
    using Restriction = R;
}
```

With SFINEA we could then do
```cpp
template <uint64_t N>
inline string<N> FromImpl<std::string, Lossy<string<N>, TruncateToCapacity_t>>::fromImpl(const std::string& value)
```

There would also be an implementation without the `Lossy<D, R>` with a `static_assert` to point to the intended implementation.

This would also work for other types, e.g. `std::vector` to `cxx::vector`, etc.

Like I said before, nothing for this PR but we should try to make it more visible to the user that something unexpected might happen.